feat: Uses new table component in Drill to Detail

michael-s-molina opened the pull request:
### SUMMARY
Uses the new table component in Drill to Detail to benefit from virtualization and deal with tables with many columns.

It also extends the new Table component with new cell and header renderers.

@eric-briscoe 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/204310917-784885ee-51bd-4db9-a1e2-ad34b111f516.mov

https://user-images.githubusercontent.com/70410625/204310446-c28a3873-3231-4e21-8e57-d8a741755c2a.mov

### TESTING INSTRUCTIONS
- Check that the pagination is working as intended
- Check that the columns are correctly formatted
- Check that you can select formatting for time columns

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

michael-s-molina committed:
feat: Uses virtualized table in Drill to Detail

michael-s-molina commented:
@eric-briscoe I noticed an error in the browser console when paginating. It seems an internal table error. Can you check?

eric-briscoe commented on superset-frontend/src/components/Table/cell-renderers/TimeCell/TimeCell.test.tsx:
```diff
@@ -0,0 +1,39 @@
+  expect(screen.getByText('2022-01-01')).toBeInTheDocument();
+});
+
+test('renders with number', async () => {
```
@michael-s-molina can we add tests that pass a mixture of null | undefined| invalid time formats as we as mix of invalid DATE values to ensure the cell renderer does not throw runtime error and break app in case a single row has bad / missing data?  I think we want to make cell renderers fault tolerant especially when doing error prone translations like dates.  antd does not give us an error protection on cell rendering so we have to make sure our cell renderers do not throw error that will break the render loop for entire table component / app

michael-s-molina replied:
Good idea. Added ✅

eric-briscoe commented on superset-frontend/src/components/Table/cell-renderers/TimeCell/index.tsx:
```diff
@@ -0,0 +1,35 @@
+  format = TimeFormats.DATABASE_DATETIME,
+  value,
+}: TimeCellProps) {
+  const timeFormatter = getTimeFormatter(format);
+  return <span>{timeFormatter.format(value)}</span>;
```
@michael-s-molina pairing this with comment on superset-frontend/src/components/Table/cell-renderers/TimeCell/TimeCell.test.tsx for testing with bad data formats / formats value mismatch to make sure we are handling any potential errors due to bad dat values for a row

michael-s-molina replied:
👍🏼

eric-briscoe reviewed (approved):
Overall looks great!  I left one set of comments about testing bad values for data cell which I think would be worthwhile.

michael-s-molina committed:
Adds more tests

geido commented on superset-frontend/src/components/Chart/DrillDetail/DrillDetailPane.tsx:
```diff
@@ -89,30 +102,64 @@ export default function DrillDetailPane({
+                { label: t('Formatted value'), value: 'formatted' },
+              ]}
+              value={
+                timeFormatting[column] === 'original' ? 'original' : 'formatted'
```
What do you think about using an Enum for these two values?

michael-s-molina replied:
Done ✅

geido commented:
/testenv up

geido commented:
/testenv up FEATURE_DRILL_TO_DETAIL=true

geido reviewed (approved):
LGMT! Just two nits

We can improve the spacing
![Screenshot 2022-11-29 at 18 06 13](https://user-images.githubusercontent.com/60598000/204592539-cf429676-775a-43c3-967c-5d3074d3cf42.png)

codyml reviewed (commented):
This looks great!  One issue I'm seeing is that the fixed-height table doesn't play well with the adjustable-height modal:

<img width="1911" alt="Screen Shot 2022-11-29 at 12 37 27 PM" src="https://user-images.githubusercontent.com/13007381/204631878-2ff019ea-c384-48f9-ac1c-bf07457ae23e.png">
<img width="1867" alt="Screen Shot 2022-11-29 at 12 38 25 PM" src="https://user-images.githubusercontent.com/13007381/204631897-4e9e9ee4-16cd-4f78-8bff-4b60a04b9f18.png">

Is there any way to make the table adjustable height, so it fills the modal vertically when expanded and leaves the pagination controls visible when minimizing?  If not, should we maybe make the modal height fixed to match the fixed height of the table?

michael-s-molina committed:
Makes table resizable

michael-s-molina commented:
@geido @codyml I addressed your comments in the last commit.

https://user-images.githubusercontent.com/70410625/204849854-e9a7b678-c64a-4df3-a7d7-4b7f675f9365.mov
